doc/tutorial/code/solutions/EtM.CPA.fst

```diff
@@ -71,9 +71,12 @@ val encrypt: k:key -> m:msg -> ST cipher
 let encrypt k m =
   recall k.log;
   let iv = random ivsize in
-  let text = if ind_cpa then createBytes (length m) 0z else repr m in
+  assume (UInt.fits (length m) 32);
+  let c0 = create (UInt32.uint_to_t (length m)) 0z in
+  let text = if ind_cpa then c0 else repr m in
   let c = CoreCrypto.block_encrypt AES_128_CBC k.raw iv text in
-  let c = iv@|c in
+  assume (UInt.fits (Bytes.length iv + Bytes.length c) 32);
```

Comment:
Unavoidable because `block_encrypt` is underspecified.
Maybe replace with an assumption on `Bytes.length c` plus an assumption on `Bytes.length iv`.